src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1086,6 +1086,46 @@ public void TestConnectionStringEnvVarHasHighestPrecedence()
             }
         }
 
+        /// <summary>
+        /// Test to validate that when an entity which will return a paginated response is queried and a custom rest base route is configured in the runtime configuration,
+        /// then the generated nextLink in the response would contain the rest base-route just before the rest path. For the subsequent query, the rest base-route will be trimmed
+        /// by the upstream before the request lands at DAB.
+        /// </summary>
+        [TestMethod]
+        public async Task TestRestBaseRouteInNextLinkForPaginatedRestResponse()
+        {
+            const string CUSTOM_CONFIG = "custom-config.json";
+            string restBaseRoute = "/rest-api";
+            TestHelper.ConstructNewConfigWithCustomSettings(CUSTOM_CONFIG, TestCategory.MSSQL, restBaseRoute: restBaseRoute);
+            string[] args = new[]
+            {
+                    $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                string requestPath = "/api/MappedBookmarks";
+                HttpMethod httpMethod = SqlTestHelper.ConvertRestMethodToHttpMethod(RestMethod.Get);
+                HttpRequestMessage request = new(httpMethod, requestPath);
+
+                HttpResponseMessage response = await client.SendAsync(request);
+                string responseBody = await response.Content.ReadAsStringAsync();
+                Assert.IsTrue(response.StatusCode is HttpStatusCode.OK);
+
+                JsonElement responseElement = JsonSerializer.Deserialize<JsonElement>(responseBody);
```

Comment:
Should we create a C# type for the response? We know what it'd look like so would that make sense, rather than us having to walk JSON structures in the test?

Reply:
I don't think we should create it as yet. If the use cases increase we can create a c# type. Do you see other use cases currently as well? I might be missing..

Reply:
It's more looking at the complexity of the following `Assert` statements and that we need to have the test aware of the structure of the JSON payload its working with. While it's not inherently problematic, I always ask the question when we have to walk JSON structures in tests.

Reply:
Will take this into consideration in a follow up issue.